specification/dartLangSpec.tex

```diff
@@ -4326,6 +4326,9 @@ \section{Interfaces}
 (\ref{bindingActualsToFormals}),
 where $T_0$ is a class with interface $I$,
 is also considered to have interface $I$.
+Similarly, when $T$ is $F$ bounded where $F$ is a function type,
```

Comment:
Is "$F$ bounded" a term we generally use for a type variable having an upper bound of `$F$`.
I feel it's very confusable with having F-bounded polymorphism. 😉 

I can see we use "any type $T$ which is $T_0$ bounded`" above. Does it just mean "which has $T_0$ as supertype"? Or must $T$ be a type variable and $T_0$ its bound?
In the former case, it's not true, we need the most specific function super-type for the type of `call`, not any function super-type.

Reply:
Yes, it's a bit of a conflict to use `$F$` to name the type, but function types are often named like that. We do use `$T$ bounded` to mean types like `X extends T` and `X extends Y`, `Y extends T`, with special cases like 'function-type bounded' as a common term for all `$F$` bounded types where `$F$` is a function type. I changed the name to `$T_0$` (that's consistent with the previous sentence, so at least it's consistent with _something_ ;-).

We can have cases where `$T$` is `$T_0$` bounded and `$T$` isn't a type variable: For instance, `$T_0$` is `$T_0$` bounded for all `$T_0$`. But whenever we _use_ "`$T_0$ bounded", `$T_0$` is a less general type: It can be an interface type or a function type, and we might need additional cases in the future. But the point is that we just want to skip over the `extends` steps for a type variable until we get to a type that has some affordances, and then we'll give the same affordances to that type variable.

The notion of being `$T_0$` bounded allows us to skip from a type variable to its bound until we encounter a type which is not a type variable, but it does not allow any other subtype steps. So there should not be a problem with the most specific function supertype vs. other function types that are supertypes of the most specific one.